src/System.Linq/src/System/Linq/Count.cs

```diff
@@ -15,6 +15,8 @@ public static int Count<TSource>(this IEnumerable<TSource> source)
             if (source == null) throw Error.ArgumentNull("source");
             ICollection<TSource> collectionoft = source as ICollection<TSource>;
             if (collectionoft != null) return collectionoft.Count;
+            IIListProvider<TSource> listProv = source as IIListProvider<TSource>;
+            if (listProv != null) return listProv.GetCount(false);
```

Comment:
Nit: would you mind using a named argument at the call site so that it's clear what the Boolean means?